### ISSUE TITLE
@ref links are incorrect when used in class methods body

### ISSUE THREAD

**Issue by AlexTr** [doxygen/doxygen#10869]
**Describe the bug**
When \ref is used inside class method, it generates link to source code files which are actually missing, the same \ref in standalone function body generates correct links

**Screenshots**
<img width="1463" alt="Screenshot 2024-05-16 at 20 31 48" src="https://github.com/doxygen/doxygen/assets/3324866/5319fc22-d4c7-41d4-8390-861895b0a3ee">

**To Reproduce**
Create file with the following contents:
```php
<?php
/**
 * @defgroup    CoreClasses Core Classes
 * @{
 */

/**
 * Sample class
 */
class FooClass {

    /** 
     * @anchor anc_foo_method_with_desc_above
     * Sample class method
     * @reminder @ref anc_foo_method_with_desc_above
     */
    public function foo_method_with_desc_above() {
        return 111;
    }

    public function foo_method_with_desc_below() {
        /**
         * @anchor anc_foo_method_with_desc_below
         * Sample class method
         * @reminder @ref anc_foo_method_with_desc_below
         */
        return 222;
    }

}

/** @} */
```
Run Doxygen 1.10.0 with the following config:
```
PROJECT_NAME           = BUGREPORT2
GENERATE_TREEVIEW      = YES
ALIASES                = "reminder=\xrefitem reminder \"REMINDER\" \"Reminders index\""
GENERATE_LATEX         = NO
```

**Expected behavior**
When method/function description is in function/body it should be treated the same way as in front of function/method definition and to work the same way, link should be correct and refer to the specified anchor.

**Version**
Ubuntu 20.04
Doxygen 1.10.0 (with oloder version the behaviour is the same)

**Additional context**
Minimal reproducible example is also attached.
[test.tgz](https://github.com/doxygen/doxygen/files/15336953/test.tgz)

**Comment by albert-github** [comment]
Difficult problem as we have here to do with "internal" documentation.
The problem can also be shown by means of the smaller example:
```
/// \file

void foo_method_with_desc_inside() {
    /**
     * @anchor anc_foo_method_with_desc_inside
     * Sample class method
     * @ref anc_foo_method_with_desc_inside
     */
}
/**
 * @anchor anc_foo_method_with_desc_outside
 * Sample class method
 * @ref anc_foo_method_with_desc_outside
 */
void foo_method_with_desc_outside() {
}
```
and when running `doxygen -d  Sections` we get:
```
Section = anc_foo_method_with_desc_inside, file = .../FooClass.h, title = , type = 7, ref =
Section = anc_foo_method_with_desc_outside, file = _foo_class_8h, title = , type = 7, ref =
```
so we see that the file / anchor (`type = 7`) for the inside function is not "translated".
At first sight this would be according to the idea of "internal" documentation, though adding the `\internal` command and the setting `INTERNAL_DOCS=YES`  still gives the same problem!

We also have the strange phenomenon that there is no warning like:
```
.../FooClass.h:3: warning: Member foo_method_with_desc_inside() (function) of file FooClass.h is not documented.
```
that should be present when there is no documentation for the function.

So I think the problem is that the `\anchor` command was not foreseen in "internal" documentation and thus is not handled (yet).

Looks like relevant type of functions are:
- `findSectionsInDocumentation` in different files (`*def.cpp`)
- `docFindSections` in different files (`*def.cpp` and `docparser.cpp`)
- `DocTokenizer::findSections`

@doxygen any idea on how to tackle this problem?

Small example: [example.tar.gz](https://github.com/doxygen/doxygen/files/15346906/example.tar.gz)

**Comment by AlexTr** [comment]
Thank you @albert-github for taking care of this issue, it looks like you are correct with you assumption about `\anchor` command, also there is another issue with the same problem - https://github.com/doxygen/doxygen/issues/9663

**Comment by doxygen** [comment]
@AlexTr Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by AlexTr** [comment]
@doxygen  Thank you a lot for quick fix 🙏 it works correctly now!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
